react/Table/index.jsx

```diff
@@ -1,39 +1,69 @@
 import React from 'react'
-import styles from './styles.styl'
 import cx from 'classnames'
+import PropTypes from 'prop-types'
+
+import styles from './styles.styl'
 
-export const Table = props => {
+export const Table = React.forwardRef((props, ref) => {
   const { className, ...rest } = props
 
-  return <div className={cx(styles.Table, className)} {...rest} />
-}
+  return <div className={cx(styles.Table, className)} {...rest} ref={ref} />
+})
 
-export const TableHead = props => {
+export const TableHead = React.forwardRef((props, ref) => {
   const { className, ...rest } = props
 
-  return <div className={cx(styles.TableHead, className)} {...rest} />
-}
+  return <div className={cx(styles.TableHead, className)} {...rest} ref={ref} />
+})
 
-export const TableBody = props => {
+export const TableBody = React.forwardRef((props, ref) => {
   const { className, ...rest } = props
 
-  return <div className={cx(styles.TableBody, className)} {...rest} />
-}
+  return <div className={cx(styles.TableBody, className)} {...rest} ref={ref} />
+})
 
-export const TableRow = props => {
+export const TableRow = React.forwardRef((props, ref) => {
   const { className, ...rest } = props
 
-  return <div className={cx(styles.TableRow, className)} {...rest} />
-}
+  return <div className={cx(styles.TableRow, className)} {...rest} ref={ref} />
+})
 
-export const TableCell = props => {
+export const TableCell = React.forwardRef((props, ref) => {
   const { className, ...rest } = props
 
-  return <div className={cx(styles.TableCell, className)} {...rest} />
-}
+  return <div className={cx(styles.TableCell, className)} {...rest} ref={ref} />
+})
 
-export const TableHeader = props => {
+export const TableHeader = React.forwardRef((props, ref) => {
   const { className, ...rest } = props
 
-  return <div className={cx(styles.TableHeader, className)} {...rest} />
+  return (
+    <div className={cx(styles.TableHeader, className)} {...rest} ref={ref} />
+  )
+})
+
+Table.propTypes = {
+  className: PropTypes.string,
+  ref: PropTypes.oneOfType([PropTypes.func, PropTypes.object])
```

Comment:
I do not think that we should put `ref` in the props, nor className in the propType. It should be a given that className / style and ref are supported by our components, no ? It oculd be a mention at the top of the styleguide, I can't find the page on material UI that said the same thing though.

Reply:
If all our components support className, style and ref I will be agreed with you. Since they don't, I find it useful to show the fact they support it. 

Maybe the real fix is to pass on all our components and fix this issue, but it's a bit of work I don't want to do right now.

Reply:
Why not putting, "all our components should support className / style and ref", please open an issue if you find a component that does not conform to do this ?

Reply:
I will. We did it for this one, and we'll do it again for others. But instead of relying on "external reports", I want to take the time to pass on all our components to check if they support those props or not before writing somewhere that our components handle "className, style & ref" without any issues.

Reply:
OK.